autopilot/migrator.go

```diff
@@ -130,6 +130,12 @@ func (m *migrator) performMigrations(p *workerPool, set string) {
 
 OUTER:
 	for {
+		// recompute health first.
+		if err := b.RecomputeHealth(ctx); err != nil {
+			m.logger.Errorf("failed to recompute cached health before migration", err)
+			return
```

Comment:
We could drop the return here so an error in `RecomputeHealth` doesn't prevent migrations from taking place at least once? It doesn't matter that much though but maybe we can register a critical alert? I feel this is one of those spots where migrations simply don't happen for some reason and we don't find out about it until it's too late.